File: app/src/main/java/org/fossasia/susi/ai/rest/model/Action.java
package org.fossasia.susi.ai.rest.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Action {

    @SerializedName("delay")
    @Expose
    private long delay;
    @SerializedName("expression")
    @Expose
    private String expression;
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("link")
    @Expose
    private String anchorLink;
    @SerializedName("text")
    @Expose
    private String anchorText;
    @SerializedName("query")
    @Expose
    private String query;

    public String getExpression() {
        return expression;
    }

    public String getType() {
        return type;
    }

    public long getDelay() {
        return delay;
    }

    public String getAnchorLink() {
        return anchorLink;
    }

    public String getAnchorText() {
        return anchorText;
    }

    public String getQuery() {
        return query;
    }
}

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/Action.kt
package org.fossasia.susi.ai.rest.responses.susi

import com.google.gson.annotations.SerializedName

data class Action(
    val delay: Long = 0,
    val expression: String = "",
    val type: String = "",
    @SerializedName("link")
    val anchorLink: String? = null,
    @SerializedName("text")
    val anchorText: String? = null,
    val query: String = "",
    val latitude: Double = 0.toDouble(),
    val longitude: Double = 0.toDouble(),
    val zoom: Double = 0.toDouble(),
    val count: Int = 0,
    val language: String = "en",
    var columns: Map<String, Any>? = null,
    var identifier: String = "",
    var plan_delay: Long = 0
)
